### PR TITLE
Clean up configuration

### PR DESCRIPTION
**Pull request by willkg**
This cleans up a bunch of configuration issues. Figured I'd break this out into a separate PR rather than do it along with the boto -> boto3 changes.

None of these configuration changes require changes to the server configuration.

### CODE REVIEW

File: docker/config/local_dev.env [hunk 1]
```diff
@@ -35,7 +35,7 @@ resource.metrics.markus_backends=markus.backends.logging.LoggingMetrics,markus.b
 # secrets.sentry.dsn=
 
 # pubsub
---------
+# ------
```
Comment (willkg) [inline docker/config/local_dev.env]: This was a mistake. I have no idea what `---------` does in an env file.

File: docker/config/local_dev.env [hunk 2]
```diff
@@ -85,22 +85,17 @@ processor.command_pathname=/stackwalk/stackwalker
 
 # Set the telemetry bucket name explicitly
 destination.telemetry.bucket_name=telemetry_bucket
+telemetry.bucket_name=telemetry_bucket
```
Comment (willkg) [inline docker/config/local_dev.env]: The `socorro/external/boto/upload_telemetry_schema.py` configman app uses `telemetry.bucket_name` for the name of the bucket to push the schema to. This wasn't set in the local dev environment.

It is set in the server environments, so this change doesn't affect the servers at all.

File: docker/config/local_dev.env [hunk 3]
```diff
@@ -85,22 +85,17 @@ processor.command_pathname=/stackwalk/stackwalker
...
 
 # webapp
 # ------
 
 ALLOWED_HOSTS=localhost,webapp
-AWS_HOST=localstack-s3
-AWS_PORT=5000
-AWS_ACCESS_KEY=foo
-AWS_SECRET_ACCESS_KEY=foo
```
Comment (willkg) [inline docker/config/local_dev.env]: None of the `AWS*` settings are used in the webapp code. I removed them from here and the configuration files.

File: docker/config/local_dev.env [hunk 4]
```diff
@@ -85,22 +85,17 @@ processor.command_pathname=/stackwalk/stackwalker
...
 CACHE_LOCATION=memcached:11211
 DATABASE_URL=postgres://postgres:aPassword@postgresql:5432/breakpad
-ELASTICSEARCH_URLS=http://elasticsearch:9200
```
Comment (willkg) [inline docker/config/local_dev.env]: This wasn't in use in the webapp code.

File: docker/config/local_dev.env [hunk 5]
```diff
@@ -85,22 +85,17 @@ processor.command_pathname=/stackwalk/stackwalker
...
 SENTRY_DSN=
 SECRET_KEY=secretkey
 STATSD_HOST=statsd
-TELEMETRY_BUCKET_NAME=telemetry_bucket
```
Comment (willkg) [inline docker/config/local_dev.env]: This isn't used in the webapp code.

File: docker/config/never_on_a_server.env [hunk 6]
```diff
@@ -21,7 +21,6 @@ DEBUG=True
 STATIC_ROOT=/tmp/crashstats-static/
 
 # For connecting to local S3, we need to connect insecurely.
-AWS_SECURE=False
```
Comment (willkg) [inline docker/config/never_on_a_server.env]: This isn't used in the webapp code.

File: docker/config/test.env [hunk 7]
```diff
@@ -10,12 +10,11 @@ STATSD_CLIENT=django_statsd.clients.null
...
 
 # Boto
-AWS_ACCESS_KEY=something
-AWS_SECRET_ACCESS_KEY=something
 resource.boto.calling_format=boto.s3.connection.SubdomainCallingFormat
 resource.boto.resource_class=socorro.external.boto.connection_context.RegionalS3ConnectionContext
 resource.boto.bucket_name=crashstats-test
-TELEMETRY_BUCKET_NAME=telemetry-test
+destination.telemetry.bucket_name=telemetry-test
+telemetry.bucket_name=telemetry-test
```
Comment (willkg) [inline docker/config/test.env]: These two were missing from the test configuration. Currently, the test code hard-codes the bucket name to use and all the S3 stuff is mocked out, so this doesn't affect anything ... yet.

File: requirements/default.txt [hunk 8]
```diff
@@ -139,9 +139,6 @@ urlwait==0.4 \
 pytest==5.2.2 \
     --hash=sha256:27abc3fef618a01bebb1f0d6d303d2816a99aa87a5968ebc32fe971be91eb1e6 \
     --hash=sha256:58cee9e09242937e136dbb3dab466116ba20d6b7828c7620f23947f37eb4dae4
-pytest-wholenodeid==0.2 \
-    --hash=sha256:6bc4edaf560c8b45ac4669bb8b99d8fbbb7f6e5af8dc7727200cd4d3ef4668c2 \
-    --hash=sha256:bdcb4df88ca182a19752ef95150c0881deaafc2c593ef67c61515894e5309206
```
Comment (willkg) [inline requirements/default.txt]: I'm no longer maintaining this library. It also has some minor issues with it for certain kinds of test definitions--the details escape me. Anyhow, time to stop using it.

File: webapp-django/crashstats/settings/base.py [hunk 9]
```diff
@@ -572,7 +576,7 @@ def filter(self, record):
 implementations_config = dj_database_url.parse(implementations_database_url)
 
 # Config for when the models pull directly from socorro.external classes.
-SOCORRO_IMPLEMENTATIONS_CONFIG = {
```
Comment (willkg) [inline webapp-django/crashstats/settings/base.py]: This name is hard to use, so I shortened it.

File: webapp-django/crashstats/settings/base.py [hunk 10]
```diff
@@ -631,7 +633,7 @@ def filter(self, record):
             ),
         },
     },
-    "telemetrydata": {"bucket_name": config("TELEMETRY_BUCKET_NAME", None)},
+    "telemetrydata": {"bucket_name": config("destination.telemetry.bucket_name", None)},
```
Comment (willkg) [inline webapp-django/crashstats/settings/base.py]: `destination.telemetry.bucket_name` is what the processor uses. Given that we're using Socorro configman names in other places, I figured we should change this one, too. Fewer places to have to change things in the future.

File: docker/config/local_dev.env (hunk 5 above)
Comment (willkg) [reply docker/config/local_dev.env]: I mispoke. I changed this setting to pull from `destination.telemetry.bucket_name`.

**Comment by willkg** [comment]
@jwhitlock Given that this is fiddling with configuration, it's worth double-checking.

File: docker/config/never_on_a_server.env (hunk 6 above)
Comment (jwhitlock) [reply docker/config/never_on_a_server.env]: This is set in the settings, but not used in the code, and maybe should be removed as well:

https://github.com/mozilla-services/socorro/blob/c3c44333af0118f60670da10d8709010905c71eb/webapp-django/crashstats/settings/base.py#L567

**Comment by willkg** [comment]
Whoops--I missed a file. Bah.

**Comment by willkg** [comment]
^^^ The `AWS_*` settings are no longer in `webapp-django/crashstats/settings/base.py`.

**Comment by willkg** [comment]
Thank you!

I'll wait for this to deploy to stage and then verify things are working.

**Pull request merged**